apps/api/src/lib/id-generation/types/UuidDocument.ts

```diff
@@ -0,0 +1,4 @@
+export type UuidDocument<TIdFormat> = {
+    id: TIdFormat;
+    isAvailable: boolean;
```

Comment:
Let's not go there now, but here's an interesting idea. We could instead have
```ts
isUsedBy: AggregateCompositeIdentifier; // e.g. `song #123`, `note #55` 
```
with `isUsedBy` either null \ undefined or set to a special string (Symbol in memory).
Then when we `use` the ID, we put the composite identifier of the aggregate that uses that ID. In this way, any `UUID`
would point you back to the specific doc in the specific collection with one lookup in our Aggregate IDs table.

At least make a note about this.

Reply:
done